src/services/practice-programs/PracticePrograms.queries.ts

```diff
@@ -1,26 +1,48 @@
-import { useQuery } from "@tanstack/react-query";
-import { PaginatedEntity } from "../../common/interfaces/PaginatedEntity.interface";
-import { searchPracticePrograms } from "./PracticePrograms.service";
+import { useQuery } from '@tanstack/react-query';
+import { PaginatedEntity } from '../../common/interfaces/PaginatedEntity.interface';
+import { IPracticeProgram } from '../../common/interfaces/PracticeProgram.interface';
+import { usePracticePrograms } from '../../store/Selectors';
+import useStore from '../../store/Store';
+import { searchPracticePrograms } from './PracticePrograms.service';
+
+export const usePracticeProgramsQuery = () => {
+  const { setPracticePrograms } = useStore();
+  const {
+    filters: { search, locationId, faculties, workingHours, domains, start, end },
+    meta: { currentPage, itemsPerPage },
+  } = usePracticePrograms();
 
-export const usePracticeProgramsQuery = (
-  limit: number,
-  page: number,
-  search?: string,
-  locationId?: number,
-  faculties?: number[],
-  workingHours?: string,
-  domains?: number[],
-  start?: string,
-  end?: string,
-) => {
   return useQuery(
-    ['practice-programs', limit, page, search, locationId, faculties, workingHours, domains, start, end],
-    () => searchPracticePrograms(limit, page, search, locationId, faculties, workingHours, domains, start, end),
+    [
+      'practice-programs',
+      itemsPerPage,
+      currentPage,
+      search,
+      locationId,
+      faculties,
+      workingHours,
+      domains,
+      start,
+      end,
+    ],
+    () =>
+      searchPracticePrograms(
+        itemsPerPage,
+        currentPage,
+        search,
+        locationId,
+        faculties,
+        workingHours,
+        domains,
+        start,
+        end,
+      ),
     {
-      onSuccess: (data: PaginatedEntity<any>) => {
-        console.log(data);
+      onSuccess: (data: PaginatedEntity<IPracticeProgram>) => {
+        setPracticePrograms(data);
       },
-      enabled: false,
+      enabled: !!(currentPage && itemsPerPage),
```

Comment:
Daca e currentPage vreodata 0, e problema

Reply:
Not really, pentru ca current page trebuie sa fie >= 1 asa ca daca este 0 nu cere date lucru pe care il vrem